Source/NSCollectionView.m

```diff
@@ -1114,4 +1507,591 @@ - (BOOL) wantsPeriodicDraggingUpdates
+      [self setFrame: newRect];
+      for (cs = 0; cs < ns; cs++)
+        {
+          [self _loadSectionAtIndex: cs];
+        }
+      [self _updateParentViewFrame];
+      [self setPostsFrameChangedNotifications: f]; // reset
+      _allowReload = YES;
+    }
+}
+
+- (void) reloadSections: (NSIndexSet *)sections
+{
+  NSUInteger *buffer = NULL;
+  NSUInteger c = 0;
+  NSUInteger i = 0;
+  
+  c = [sections getIndexes: buffer
+                  maxCount: [sections count]
+              inIndexRange: NULL];
+
+  if (buffer != NULL)
+    {
+      for (i = 0; i < c; i++)
+        {
+          NSUInteger cs = buffer[i];
+          [self _loadSectionAtIndex: cs];
+        }
+    }
+}
+
+- (void) reloadItemsAtIndexPaths: (NSSet *)indexPaths
+{
+  FOR_IN(NSIndexPath*, p, indexPaths)
+    {
+      [self _loadItemAtIndexPath: p];
+    }
+  END_FOR_IN(indexPaths);
+}
+
+/* Prefetching Collection View Cells and Data */
+
+- (id<NSCollectionViewPrefetching>) prefetchDataSource
+{
+  return _prefetchDataSource;
+}
+
+- (void) setPrefetchDataSource: (id<NSCollectionViewPrefetching>)prefetchDataSource
+{
+  _prefetchDataSource = prefetchDataSource;
+}
+
+/* Getting the State of the Collection View */
+
+- (NSInteger) numberOfSections
+{
+  NSInteger n = 0;
+
+  if ([_dataSource respondsToSelector: @selector(numberOfSectionsInCollectionView:)])
+    {
+      n = [_dataSource numberOfSectionsInCollectionView: self];
+    }
+  
+  return n;
+}
+
+- (NSInteger) numberOfItemsInSection: (NSInteger)section
+{
+  // Since this is a required method by the delegate we can assume it's presence
+  // if it is not there, tests on macOS indicate that an unrecognized selector
+  // exception is thrown.
+  return [_dataSource collectionView: self numberOfItemsInSection: section];
+}
+
+/* Inserting, Moving and Deleting Items */
+
+- (void) insertItemsAtIndexPaths: (NSSet *)indexPaths
+{
+}
+
+- (void) moveItemAtIndexPath: (NSIndexPath *)indexPath 
+                 toIndexPath: (NSIndexPath *)newIndexPath
+{
+}
+
+- (void) deleteItemsAtIndexPaths: (NSSet *)indexPaths
+{
+}
+
+/* Inserting, Moving, Deleting and Collapsing Sections */
+
+- (void) insertSections: (NSIndexSet *)sections
+{
+}
+
+- (void) moveSection: (NSInteger)section 
+           toSection: (NSInteger)newSection
+{
+}
+
+- (void) deleteSections: (NSIndexSet *)sections
+{
+}
+
+// 10.12 method...
+
+- (IBAction) toggleSectionCollapse: (id)sender
+{
+}
+
+// 10.11 methods...
+
+- (BOOL) allowsEmptySelection
+{
+  return _allowsEmptySelection;
+}
+
+- (void) setAllowsEmptySelection: (BOOL)flag;
+{
+  _allowsEmptySelection = flag;
+}
+
+- (IBAction) selectAll: (id)sender
+{
+  NSMutableSet *paths = [NSMutableSet setWithCapacity: [_itemsToIndexPaths count]];
+
+  FOR_IN(NSCollectionViewItem*, obj, _itemsToIndexPaths)
+    {
+      if ([obj respondsToSelector: @selector(setSelected:)])
+        {
+	  NSIndexPath *p = [_itemsToIndexPaths objectForKey: obj];
+
+          [obj setSelected: YES];
```

Comment:
The actual selection already gets done in-setSelectionIndexPaths:, you don't need to do it here. But this is no big issue it would just make the code as a whole easier to understand if everything builds up on minimal methods.

Reply:
I removed these.  This should be okay now.